src/uu/ls/src/dired.rs

```diff
@@ -179,6 +179,16 @@ pub fn update_positions(dired: &mut DiredOutput, start: usize, end: usize) {
     dired.padding = 0;
 }
 
+/// Checks if the "--dired" or "-D" argument is present in the command line arguments.
+/// we don't use clap here because we need to know if the argument is present
+/// as it can be overridden by --hyperlink
+pub fn is_dired_arg_present() -> bool {
+    let args: Vec<String> = std::env::args().collect();
+    let dired_index = args.iter().position(|x| x == "--dired" || x == "-D");
+
+    dired_index.is_some()
```

Comment:
As you are only interested in the presence of `--dired`/`-D` but not the position, you can simplify it to:
```suggestion
    args.iter().any(|x| x == "--dired" || x == "-D")
```